minio/api.py

```diff
@@ -1044,6 +1044,7 @@ def fget_object(self, bucket_name, object_name, file_path,
 
         # Write to a temporary file "file_path.part.minio" before saving.
         tmp_file_path = tmp_file_path or f"{file_path}.{stat.etag}.part.minio"
+        makedirs(os.path.dirname(tmp_file_path)) # prepare temporary file's directory
```

Comment:
I don't think it is a good idea SDK to create entire directory tree. It is user's responsibility to have correct `file_path`. BTW this is not temporary directory rather `minio-py` creates temporary file under the directory.

Reply:
`file_path` is correct and exists, the problem is `{stat.etag}` contains a `/`, that make it become a sub-directory that doesn't exists.

<img width="746" alt="CleanShot 2023-09-19 at 09 49 53@2x" src="https://github.com/minio/minio-py/assets/4532436/9afef27f-9aff-42a6-9aa3-ea6ac33bfd18">

Reply:
Send a fix to sanitize `etag` here using `helpers.queryencode()`

Reply:
done